pkg/deployments/tasks/20230314-batch-relayer-v5/test/tetuWrapping.fork.ts

```diff
@@ -0,0 +1,132 @@
+import hre from 'hardhat';
+import { expect } from 'chai';
+import { BigNumber, Contract } from 'ethers';
+
+import { BigNumberish } from '@balancer-labs/v2-helpers/src/numbers';
+
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+
+import { describeForkTest, impersonate, getForkedNetwork, Task, TaskMode } from '../../../src';
+
+describeForkTest('TetuWrapping', 'polygon', 37945364, function () {
+  let task: Task;
+  let relayer: Contract, library: Contract;
+  let vault: Contract, authorizer: Contract;
+
+  const USDT = '0xc2132D05D31c914a87C6611C10748AEb04B58e8F';
+  const USDT_HOLDER = '0xf977814e90da44bfa03b6295a0616a897441acec';
+  const xUSDT = '0xE680e0317402ad3CB37D5ed9fc642702658Ef57F';
+
+  const TETU_GOVERNANCE = '0xcc16d636dD05b52FF1D8B9CE09B09BC62b11412B';
+  const TETU_CONTROLLER = '0x6678814c273d5088114B6E40cC49C8DB04F9bC29';
+
+  let usdtToken: Contract, tetuVault: Contract;
+  let sender: SignerWithAddress;
+  let chainedReference: BigNumber;
+  const amountToWrap = 100e6;
+
+  before('run task', async () => {
+    task = new Task('20230314-batch-relayer-v5', TaskMode.TEST, getForkedNetwork(hre));
+    await task.run({ force: true });
+    library = await task.deployedInstance('BatchRelayerLibrary');
+    relayer = await task.instanceAt('BalancerRelayer', await library.getEntrypoint());
+  });
+
+  before('load vault and tokens', async () => {
+    const vaultTask = new Task('20210418-vault', TaskMode.READ_ONLY, getForkedNetwork(hre));
+
+    vault = await vaultTask.instanceAt('Vault', await library.getVault());
+    authorizer = await vaultTask.instanceAt('Authorizer', await vault.getAuthorizer());
+  });
+
+  before('approve relayer at the authorizer', async () => {
+    const relayerActionIds = await Promise.all(
+      ['swap', 'batchSwap', 'joinPool', 'exitPool', 'setRelayerApproval', 'manageUserBalance'].map((action) =>
+        vault.getActionId(vault.interface.getSighash(action))
+      )
+    );
+
+    // We impersonate an account with the default admin role in order to be able to approve the relayer. This assumes
+    // such an account exists.
+    const admin = await impersonate(await authorizer.getRoleMember(await authorizer.DEFAULT_ADMIN_ROLE(), 0));
+
+    // Grant relayer permission to call all relayer functions
+    await authorizer.connect(admin).grantRoles(relayerActionIds, relayer.address);
+  });
+
+  before(async () => {
+    usdtToken = await task.instanceAt('IERC20', USDT);
+    tetuVault = await task.instanceAt('ITetuSmartVault', xUSDT);
+    sender = await impersonate(USDT_HOLDER);
+
+    // Set white list approvals for the batch relayer to interacte with the Tetu Smart Vault
+    const governance = await impersonate(TETU_GOVERNANCE);
+    const controller = await task.instanceAt('ITetuController', TETU_CONTROLLER);
```

Comment:
Instead of having these be included in the tasks, we can add the ABI in the test itself, reducing the dependencies. See how it's done here: https://github.com/balancer/balancer-v2-monorepo/blob/eeb691ef904b8de4569f7bdfa06acce8c8af07cb/pkg/deployments/tasks/20220714-fee-distributor-v2/test/test.fork.ts#L187